From 1c627ce0152ab8aa316f6e2f7e46c1af16ec2b52 Mon Sep 17 00:00:00 2001
From: Xi Ruoyao <xry111@xry111.site>
Date: Sat, 21 Jun 2025 17:36:09 +0800
Subject: [PATCH 03/63] AOSCOS: loonggpu: Use ccflags-y instead of EXTRA_CFLAGS

A comment in Kbuild says "using EXTRA_CFLAGS for compatibility with
kernel older than 2.6.24" which does not make any sense for loonggpu: we
cannot run such an old kernel on LoongArch.

Remove the comment and replace EXTRA_CFLAGS with ccflags-y to fix build
on Linux >= 6.15.

Signed-off-by: Xi Ruoyao <xry111@xry111.site>
---
 Kbuild | 57 +++++++++++++++++++++++++++------------------------------
 1 file changed, 27 insertions(+), 30 deletions(-)

diff --git a/Kbuild b/Kbuild
index 2275bd2..24cd2ba 100644
--- a/Kbuild
+++ b/Kbuild
@@ -62,23 +62,20 @@ $(foreach _module, $(LG_KERNEL_MODULES), \
 
 
 #
-# Define CFLAGS that apply to all the Loongson kernel modules. EXTRA_CFLAGS
-# is deprecated since 2.6.24 in favor of ccflags-y, but we need to support
-# older kernels which do not have ccflags-y. Newer kernels append
-# $(EXTRA_CFLAGS) to ccflags-y for compatibility.
+# Define CFLAGS that apply to all the Loongson kernel modules.
 #
 
-EXTRA_CFLAGS += -I$(src)/
-EXTRA_CFLAGS += -I$(src)/include
-EXTRA_CFLAGS += -I$(src)/common/inc
-EXTRA_CFLAGS += -I$(src)/loonggpu-bridge/
-EXTRA_CFLAGS += -I$(src)/lgkcd/
-EXTRA_CFLAGS += -Wall $(DEFINES) $(INCLUDES) -Wno-cast-qual -Wno-error -Wno-format-extra-args
-EXTRA_CFLAGS += -D__KERNEL__ -DMODULE -DNVRM
-EXTRA_CFLAGS += -DLG_VERSION_STRING=\"2024.01.16\"
+ccflags-y += -I$(src)/
+ccflags-y += -I$(src)/include
+ccflags-y += -I$(src)/common/inc
+ccflags-y += -I$(src)/loonggpu-bridge/
+ccflags-y += -I$(src)/lgkcd/
+ccflags-y += -Wall $(DEFINES) $(INCLUDES) -Wno-cast-qual -Wno-error -Wno-format-extra-args
+ccflags-y += -D__KERNEL__ -DMODULE -DNVRM
+ccflags-y += -DLG_VERSION_STRING=\"2024.01.16\"
 
 ifneq ($(SYSSRCHOST1X),)
- EXTRA_CFLAGS += -I$(SYSSRCHOST1X)
+ ccflags-y += -I$(SYSSRCHOST1X)
 endif
 
 # Some Android kernels prohibit driver use of filesystem functions like
@@ -88,49 +85,49 @@ endif
 PLATFORM_IS_ANDROID ?= 0
 
 ifeq ($(PLATFORM_IS_ANDROID),1)
- EXTRA_CFLAGS += -DLG_FILESYSTEM_ACCESS_AVAILABLE=0
+ ccflags-y += -DLG_FILESYSTEM_ACCESS_AVAILABLE=0
 else
- EXTRA_CFLAGS += -DLG_FILESYSTEM_ACCESS_AVAILABLE=1
+ ccflags-y += -DLG_FILESYSTEM_ACCESS_AVAILABLE=1
 endif
 
-EXTRA_CFLAGS += -Wno-unused-function
+ccflags-y += -Wno-unused-function
 
 ifneq ($(LG_BUILD_TYPE),debug)
- EXTRA_CFLAGS += -Wuninitialized
+ ccflags-y += -Wuninitialized
 endif
 
-EXTRA_CFLAGS += -fno-strict-aliasing
+ccflags-y += -fno-strict-aliasing
 
 ifeq ($(LG_BUILD_TYPE),debug)
- EXTRA_CFLAGS += -g
+ ccflags-y += -g
 endif
 
-EXTRA_CFLAGS += -ffreestanding
+ccflags-y += -ffreestanding
 
 ifeq ($(ARCH),arm64)
- EXTRA_CFLAGS += -mgeneral-regs-only -march=armv8-a
- EXTRA_CFLAGS += $(call cc-option,-mno-outline-atomics,)
+ ccflags-y += -mgeneral-regs-only -march=armv8-a
+ ccflags-y += $(call cc-option,-mno-outline-atomics,)
 endif
 
 ifeq ($(ARCH),x86_64)
- EXTRA_CFLAGS += -mno-red-zone -mcmodel=kernel
+ ccflags-y += -mno-red-zone -mcmodel=kernel
 endif
 
 ifeq ($(ARCH),powerpc)
- EXTRA_CFLAGS += -mlittle-endian -mno-strict-align -mno-altivec
+ ccflags-y += -mlittle-endian -mno-strict-align -mno-altivec
 endif
 
-EXTRA_CFLAGS += -DLG_UVM_ENABLE
-EXTRA_CFLAGS += $(call cc-option,-Werror=undef,)
-EXTRA_CFLAGS += -DLG_SPECTRE_V2=$(LG_SPECTRE_V2)
-EXTRA_CFLAGS += -DLG_KERNEL_INTERFACE_LAYER
+ccflags-y += -DLG_UVM_ENABLE
+ccflags-y += $(call cc-option,-Werror=undef,)
+ccflags-y += -DLG_SPECTRE_V2=$(LG_SPECTRE_V2)
+ccflags-y += -DLG_KERNEL_INTERFACE_LAYER
 
 #
 # Detect SGI UV systems and apply system-specific optimizations.
 #
 
 ifneq ($(wildcard /proc/sgi_uv),)
- EXTRA_CFLAGS += -DLG_CONFIG_X86_UV
+ ccflags-y += -DLG_CONFIG_X86_UV
 endif
 
 
@@ -158,7 +155,7 @@ LG_CONFTEST_CMD := /bin/sh $(LG_CONFTEST_SCRIPT) \
 
 LG_CFLAGS_FROM_CONFTEST := $(shell $(LG_CONFTEST_CMD) build_cflags)
 
-LG_CONFTEST_CFLAGS = $(LG_CFLAGS_FROM_CONFTEST) $(EXTRA_CFLAGS) -fno-pie
+LG_CONFTEST_CFLAGS = $(LG_CFLAGS_FROM_CONFTEST) $(ccflags-y) -fno-pie
 
 LG_CONFTEST_COMPILE_TEST_HEADERS := $(obj)/conftest/macros.h
 LG_CONFTEST_COMPILE_TEST_HEADERS += $(obj)/conftest/functions.h
-- 
2.52.0

